autoload/ale/handlers/haskell.vim

```diff
@@ -52,6 +48,7 @@ function! ale#handlers#haskell#HandleGHCFormat(buffer, lines) abort
         endif
 
         call add(l:output, {
+        \   'filename': l:match[1],
```

Comment:
Is the filename here an absolute path to a file? Relative paths aren't supported. Linter callbacks should somehow resolve paths to absolute paths based on current directories, project directories, etc.

Reply:
Oh I had missed that requirement. I have updated the patch so that it uses an absolute path.